src/logic/community-voice/community-voice.ts

```diff
@@ -142,9 +147,11 @@ export async function createCommunityVoiceComponent({
       throw new CommunityVoiceChatNotFoundError(communityId)
     }
 
+    // Get the user's role in the community for both public and private communities
+    const userRole = await communitiesDb.getCommunityMemberRole(communityId, userAddress)
```

Comment:
What do you think about moving this request after the banned check? We can do it concurrently with the `getUserProfileData` call to make it quicker.